### PR TITLE
Customized Error Messages

### PR DESCRIPTION
**Pull request by colincwilliams-kcsara**
Adds support for error messages that are customized in two ways:

1. Adds support for a new, optional "supportEmail" field for organizations. This is used in error messaging to direct users how to get support.
2. Using the member provider's name in error messaging.

As a part of supporting the member provider's name, this PR enforces type safety for member provider types rather than using a bare string.

image
image
image

### CODE REVIEW

**Comment by colincwilliams-kcsara** [comment]
[Display an error if the user's email address is not found in D4H](https://trello.com/c/kE5kN5Rg/31-display-an-error-if-the-users-email-address-is-not-found-in-d4h)

File: src/app/LoginPanel.tsx [hunk 1]
```diff
@@ -39,28 +39,30 @@ export default function LoginPanel() {
...
+      let supportContact: string | undefined = undefined;
       let memberProviderName = undefined;
 
       if (organization) {
-        supportEmail = organization.supportEmail;
+        supportContact = organization.supportEmail;
         memberProviderName = MemberProviderName[organization.memberProvider];
       }
 
+      supportContact ??= "your unit's operations leader";
```
Comment (valonm) [inline src/app/LoginPanel.tsx:50]: Operations Leader seems too specific to our org structure. Maybe something more generic like system administrator?

Comment (colincwilliams-kcsara) [reply src/app/LoginPanel.tsx:50]: Operations Leader was suggested somewhere in Slack before. The problem with System Administrator is I don't think anyone would know who that is unless they're already involved in the tech side of things.

When you say "our org structure", who are you referring to? I know that applies to 4x4 and ESAR. Doesn't it apply to all units in KCSARA?

Comment (valonm) [reply src/app/LoginPanel.tsx:50]: I just don't think an operations leader is the right person to contact, and if we go outside of King county other organizations may not have that title or role

Comment (valonm) [reply src/app/LoginPanel.tsx:50]: WASART is a possible example that I could see us supporting in the nearer term. They do not have "OLs".

Comment (colincwilliams-kcsara) [reply src/app/LoginPanel.tsx:50]: Ahh and of course in my head I'm thinking "leader of operations" administratively...i.e. VPO or Operations Chair. But there's not really a standard term for that either.

I guess I can use System Administrator and just trust the user to escalate up their chain of command appropriately. Will make that change.

Comment (colincwilliams-kcsara) [reply src/app/LoginPanel.tsx]: Done.

**Review by valonm — APPROVED** [review]

**Pull request merged**